src/lib/list/list.ts

```diff
@@ -50,6 +63,347 @@ export class MdList {
+    '(focus)': '_handleFocus()',
+    '(blur)': '_handleBlur()',
+    '(click)': 'toggle()',
+    'tabindex': '-1',
+    '[attr.aria-selected]': 'selected.toString()',
+    '[attr.aria-disabled]': 'disabled.toString()',
+  },
+  templateUrl: 'list-option.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class MdListOption implements AfterContentInit, OnDestroy, Focusable {
+  private _lineSetter: MdLineSetter;
+  private _disableRipple: boolean = false;
+  private _isSelectionList: boolean = false;
+  private _selected: boolean = false;
+  /** Whether the checkbox is disabled. */
+  private _disabled: boolean = false;
+  private _value: any;
+
+  /** Whether the option has focus. */
+  _hasFocus: boolean = false;
+
+  /**
+   * Whether the ripple effect on click should be disabled. This applies only to list items that are
+   * part of a nav list. The value of `disableRipple` on the `md-nav-list` overrides this flag.
+   */
+  @Input()
+  get disableRipple() { return this._disableRipple; }
+  set disableRipple(value: boolean) { this._disableRipple = coerceBooleanProperty(value); }
+
+  @ContentChildren(MdLine) _lines: QueryList<MdLine>;
+
+  /** Whether the label should appear after or before the checkbox. Defaults to 'after' */
+
+  @Input() checkboxPosition: 'before' | 'after' = 'after';
+
+  /** Whether the checkbox is disabled. */
+  get disabled() {
+    return this._disabled;
+  }
+  @Input('disabled')
+  set disabled(value: any) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+
+  @Input('value')
+  get value() { return this._value; }
+  set value( val: any) { this._value = coerceBooleanProperty(val); }
+
+  @Input('selected')
+  get selected() { return this._selected; }
+  set selected( val: boolean) { this._selected = coerceBooleanProperty(val); }
+
+  /** Emitted when the option is focused. */
+  onFocus = new EventEmitter<MdSelectionListOptionEvent>();
+
+  /** Emitted when the option is selected. */
+  @Output() select = new EventEmitter<MdSelectionListOptionEvent>();
+
+  /** Emitted when the option is deselected. */
+  @Output() deselect = new EventEmitter<MdSelectionListOptionEvent>();
+
+  /** Emitted when the option is destroyed. */
+  @Output() destroy = new EventEmitter<MdSelectionListOptionEvent>();
+
+  constructor(private _renderer: Renderer2,
+              private _element: ElementRef,
+              @Optional() public selectionList: MdSelectionList,) {
+    this._isSelectionList = !!selectionList;
+  }
+
+
+  ngAfterContentInit() {
+    this._lineSetter = new MdLineSetter(this._lines, this._renderer, this._element);
+  }
+
+  ngOnDestroy(): void {
+    this.destroy.emit({option: this});
+  }
+
+  toggle(): void {
+    if(this._disabled == false) {
+      this._selected = !this._selected;
+      if(this._selected == true) {
+        this.selectionList.selectedOptions.select(this);
+      }else {
+        this.selectionList.selectedOptions.deselect(this);
+      }
+    }
+  }
```

Comment:
The toggle function by itself can just be
```ts
toggle() {
  this.selected = !this.selected;
  this.selectionList.selectedOption.toggle(this);
}
```

The logic that a disabled option should do anything on click should be specifically tied to the click handler rather than being in the more general `toggle()`